### ISSUE TITLE
Path should be created automatically with using `-o | -output` build flag

### ISSUE THREAD

**Issue by x80486** [vlang/v#16764]
### Describe the feature

When building a project, if the `-o` (or `-output`) flag is used with a certain non-existing path, `V` states that it does not exist.

`V` should be able to create this if it doesn't, provided the user running the process can create the same.

### Use Case

Build path(s) are never stored in `Git` (or any other VCS) repository.

I strongly believe this responsibility falls in the build (or project management) tool. Creating a sub-directory by hand is a little bit cumbersome.

This is even more visible now that `src/` is supported, so having `bin/` (or something similar) at the project's root might become the default with time — but that's only one example because in reality `-o` is agnostic and any path can be specified.

### Proposed Solution

Not sure how this could be done with the `v` tool, but in essence it should do what `mkdir --parents` does when `-o` is specified — saving the specifics for the supported platforms.

### Other Information

N/A

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 6a179a2

### Environment details (OS name and version, etc.)

OS: linux, Linux version 5.15.85-1-lts (linux-lts@archlinux) (gcc (GCC) 12.2.0, GNU ld (GNU Binutils) 2.39.0) #1 SMP Thu, 22 Dec 2022 06:22:00 +0000
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
CC version: cc (GCC) 12.2.0

getwd: /home/x80486/Workshop/v-sandbox
vmodules: /home/x80486/.vmodules
vroot: /home/x80486/.asdf/installs/v/weekly.2022.51
vexe: /home/x80486/.asdf/installs/v/weekly.2022.51/v
vexe mtime: 2022-12-25 01:19:02
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 6a179a2

Git version: git version 2.39.0
Git vroot status: weekly.2022.51 (2 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

**Comment by JalonSolov** [comment]
Unfortunately, this also leads to problems, which is why V doesn't create the path by default.

Suppose you typed out some long path, did the build, then went looking for the thing you built... and it wasn't there.  What?  Did V fail?  You try it again, and there are no error, but what you built still isn't in the path where you expected it.

Eventually, you figure out that you made a typo in the path, but instead of V complaining about that path not existing, it created it... in the wrong place.

**Comment by JalonSolov** [comment]
Also... if you create the path manually, V isn't likely to fail due to not having write permissions to the output directory.

**Comment by spytheman** [comment]
Is there an example of another compiler, that creates a non existing folder for the output?

**Comment by x80486** [comment]
The mainstream languages have conventions so the project management tools already know the convention, but `Go` will happily let you do that, and `Deno` as well — or probably they are considering it.

**Comment by spytheman** [comment]
You are right that go does it ... interesting:
```
#0 13:10:55 ᛋ master /v/vnew❱rm -rf ab/
#0 13:11:06 ᛋ master /v/vnew❱go build -o ab/cd/ef /v/misc/hw.go
#0 13:11:09 ᛋ master /v/vnew❱ls -lart ab/cd/ef 
-rwxrwxr-x 1 delian delian 1813838 Dec 27 13:11 ab/cd/ef
#0 13:11:14 ᛋ master /v/vnew❱realpath ab/cd/ef
/v/vnew/ab/cd/ef
#0 13:11:23 ᛋ master /v/vnew❱
```

**Comment by spytheman** [comment]
`go build -o ab/cd/ef/ /v/misc/hw.go`
... produces a `ab/cd/ef/` folder, then inside it, a `hw` executable too.

**Comment by spytheman** [comment]
> Eventually, you figure out that you made a typo in the path, but instead of V complaining about that path not existing, it created it... in the wrong place.

That is true, but the path where the executable will be, is visible right in the command line itself, so the confusion will be short, while the convenience of not having to rely on an external build system just for this, is very nice.
